lldb/packages/Python/lldbsuite/test/cpu_feature.py

```diff
@@ -0,0 +1,58 @@
+"""
+Platform-agnostic helper to query for CPU features.
+"""
+
+import re
+
+
+class CPUFeature:
+    def __init__(self, linux_cpu_info_flag: str, darwin_sysctl_key: str):
+        self.cpu_info_flag = linux_cpu_info_flag
+        self.sysctl_key = darwin_sysctl_key
+
+    def __str__(self):
+        return self.cpu_info_flag
+
+    def is_supported(self, triple, cmd_runner):
+        if re.match(".*-.*-linux", triple):
+            err_msg, res = self._is_supported_linux(cmd_runner)
+        elif re.match(".*-apple-.*", triple):
+            err_msg, res = self._is_supported_darwin(cmd_runner)
+        else:
+            err_msg, res = None, False
+
+        if err_msg:
+            print(f"CPU feature check failed: {err_msg}")
+
+        return res
+
+    def _is_supported_linux(self, cmd_runner):
+        cmd = "cat /proc/cpuinfo"
+        err, retcode, output = cmd_runner(cmd)
+        if err.Fail() or retcode != 0:
+            err_msg = f"cat /proc/cpuinfo failed: {output}"
+            return err_msg, False
+
+        return None, (self.cpu_info_flag in output)
+
+    def _is_supported_darwin(self, cmd_runner):
+        cmd = f"sysctl -n {self.sysctl_key}"
+        err, retcode, output = cmd_runner(cmd)
+        if err.Fail() or retcode != 0:
+            return output, False
+
+        return None, (output.strip() == "1")
+
+
+# List of CPU features
+FPMR = CPUFeature("fpmr", "???")
+GCS = CPUFeature("gcs", "???")
+LASX = CPUFeature("lasx", "???")
+LSX = CPUFeature("lsx", "???")
+MTE = CPUFeature("mte", "???")
+MTE_STORE_ONLY = CPUFeature("mtestoreonly", "???")
+PTR_AUTH = CPUFeature("paca", "hw.optional.arm.FEAT_PAuth2")
+SME = CPUFeature("sme", "hw.optional.arm.FEAT_SME")
+SME_FA64 = CPUFeature("smefa64", "???")
+SME2 = CPUFeature("sme2", "hw.optional.arm.FEAT_SME2")
+SVE = CPUFeature("sve", "???")
```

Comment:
Split these by architecture. So now it would be one for AArch64 and one for Loongarch.

Also there should be a comment with the full extension name. We can do that later though, and I can fill in the ones I added in the past.

Reply:
Also wonder if they should be namespaced. Most of the time it's obvious where a feature comes from because it's in a test file for that architecture, but generally in lldb where we have what is effectively an enum with values shared between architectures, it's got some prefix to it.

I'm sure we could do some Python magic to automatically build an object so that you can say: `cpu_feature.AArch64.SME`.

Assuming we don't care about performance here, which is likely:
```
class AArch64(object):
  @property
  def SME(self):
    return ("sme", "hw.optional.arm.FEAT_SME")
```

Reply:
Or, to save a few lines:
```
class AArch64(object):
  def __init__(self):
    self.features = {'SME' : ("sme", "hw.optional.arm.FEAT_SME"), ....}

  def __getitem__(name):
    try:
      return self.features[name]
    except KeyError:
      raise RuntimeError(f"No such AArch64 CPU feature {name}")
```
It's Python, there's 100 ways to do this but you see the end result I'm going for anyway.